### PR TITLE
Fix update

### PR DESCRIPTION
**Pull request by Fatme**
Currently we're copying only framework files. With latest version are released TNSDebugging.framework and Metadata folders as part of framework. This PR changes the logic so that we can update TNSDebugging.framework and Metadata. This PR fixes the case when the old and new xcode projects are different. Also fixes $tns platform update <platform>@<version> and $tns platform add <platform>@<version> commands.

https://github.com/NativeScript/nativescript-cli/issues/305

### CODE REVIEW

File: lib/services/ios-project-service.ts [hunk 1]
```diff
@@ -193,7 +196,44 @@ class IOSProjectService implements  IPlatformProjectService {
...
+
+		}).future<boolean>()();
+	}
+
+	public updatePlatform(currentVersion: string, newVersion: string): IFuture<void> {
+		return (() => {
+			// Copy old file to options["profile-dir"]
+			var sourceFile = path.join(this.platformData.projectRoot, util.format("%s.xcodeproj", this.$projectData.projectName));
+			var destinationFile = path.join(options.profileDir, "xcodeproj");
+			//this.$fs.copyFile(sourceFile, destinationFile).wait();
```
Comment (teobugslayer) [inline lib/services/ios-project-service.ts]: commented code, consider removing it entirely

File: lib/services/platform-service.ts [hunk 2]
```diff
@@ -442,19 +443,28 @@ export class PlatformService implements IPlatformService {
...
 
-			if(semver.gt(currentVersion, newVersion)) { // Downgrade
-				var isUpdateConfirmed = this.$prompter.confirm("You are going to update to lower version. Are you sure?", () => "n").wait();
-				if(isUpdateConfirmed) {
+				if(!semver.valid(newVersion)) {
+					this.$errors.fail("The version %s is not valid. The version should consists from 3 parts separated by dot.", newVersion);
+				}
+
+				if(semver.gt(currentVersion, newVersion)) { // Downgrade
+					var isUpdateConfirmed = this.$prompter.confirm("You are going to update to lower version. Are you sure?", () => "n").wait();
```
Comment (teobugslayer) [inline lib/services/platform-service.ts]: This message sounds weird to me. @ikoevska, any suggestions?

File: lib/services/platform-service.ts [hunk 3]
```diff
@@ -442,19 +443,28 @@ export class PlatformService implements IPlatformService {
...
+				} else if(semver.eq(currentVersion, newVersion)) {
+					this.$errors.fail("Current and new version are the same.");
+				} else {
 					this.updatePlatformCore(platformData, currentVersion, newVersion).wait();
 				}
-			} else if(semver.eq(currentVersion, newVersion)) {
-				this.$errors.fail("Current and new version are the same.");
 			} else {
-				this.updatePlatformCore(platformData, currentVersion, newVersion).wait();
+				var isUpdateConfirmed = this.$prompter.confirm(util.format("We need to override xcodeproj file. The old one will be saved at %s Are you sure?", options.profileDir), () => "y").wait();
```
Comment (teobugslayer) [inline lib/services/platform-service.ts]: missing full stop after %s

File: lib/services/platform-service.ts [hunk 4]
```diff
@@ -484,32 +514,35 @@ export class PlatformService implements IPlatformService {
...
+		}).future<any>()();
 	}
 
-	private getNpmCacheDirectory(packageName: string, version: string): IFuture<string> {
+	private ensurePackageIsCached(cachedPackagePath: string, packageName: string, version: string): IFuture<void> {
 		return (() => {
-			var npmCacheDirectoryPath = this.getNpmCacheDirectoryCore(packageName, version);
-
-			if(!this.$fs.exists(npmCacheDirectoryPath).wait()) {
+			if(!this.$fs.exists(cachedPackagePath).wait()) {
```
Comment (teobugslayer) [inline lib/services/platform-service.ts]: Consider removing this check.

Comment (Fatme) [inline lib/services/platform-service.ts]: I was thinking about it, but I can't figure out a better solution. I'm opened to any suggestions.

Comment (teobugslayer) [inline lib/services/platform-service.ts]: After talking in person, we agreed on the current code.

**Comment by teobugslayer** [comment]
:+1:

**Pull request merged**